benches/benchmarks.rs

```diff
@@ -57,6 +59,140 @@ fn setup_engine() -> EngineState {
+                    None,
+                )
+                .unwrap();
+            })
+    }
+
+    #[divan::bench(args = [1, 5, 10, 100, 1_000])]
+    fn for_range(bencher: divan::Bencher, n: i32) {
+        let mut engine = setup_engine();
+        load_standard_library(&mut engine).unwrap();
+        let commands = Spanned {
+            span: Span::unknown(),
+            item: format!("(for $x in (1..{}) {{ sleep 50ns }})", n).to_string(),
+        };
+        let stack = nu_protocol::engine::Stack::new();
+
+        bencher
+            .with_inputs(|| (engine.clone(), stack.clone()))
+            .bench_values(|(mut engine, mut stack)| {
+                evaluate_commands(
+                    &commands,
+                    &mut engine,
+                    &mut stack,
+                    PipelineData::empty(),
+                    None,
+                )
+                .unwrap();
+            })
+    }
+
+    #[divan::bench(args = [1, 5, 10, 100, 1_000])]
+    fn each(bencher: divan::Bencher, n: i32) {
+        let mut engine = setup_engine();
+        load_standard_library(&mut engine).unwrap();
+        let commands = Spanned {
+            span: Span::unknown(),
+            item: format!("(1..{}) | each {{|_| sleep 50ns }} | ignore", n).to_string(),
+        };
+        let stack = nu_protocol::engine::Stack::new();
+
+        bencher
+            .with_inputs(|| (engine.clone(), stack.clone()))
+            .bench_values(|(mut engine, mut stack)| {
+                evaluate_commands(
+                    &commands,
+                    &mut engine,
+                    &mut stack,
+                    PipelineData::empty(),
+                    None,
+                )
+                .unwrap();
+            })
+    }
+
+    #[divan::bench(args = [1, 5, 10, 100, 1_000])]
+    fn par_each_1t(bencher: divan::Bencher, n: i32) {
+        let mut engine = setup_engine();
+        load_standard_library(&mut engine).unwrap();
+        let commands = Spanned {
+            span: Span::unknown(),
+            item: format!("(1..{}) | par-each -t 1 {{|_| sleep 50ns }} | ignore", n).to_string(),
+        };
+        let stack = nu_protocol::engine::Stack::new();
+
+        bencher
+            .with_inputs(|| (engine.clone(), stack.clone()))
+            .bench_values(|(mut engine, mut stack)| {
+                evaluate_commands(
+                    &commands,
+                    &mut engine,
+                    &mut stack,
+                    PipelineData::empty(),
+                    None,
+                )
+                .unwrap();
+            })
+    }
+    #[divan::bench(args = [1, 5, 10, 100, 1_000])]
+    fn par_each_2t(bencher: divan::Bencher, n: i32) {
+        let mut engine = setup_engine();
+        load_standard_library(&mut engine).unwrap();
+        let commands = Spanned {
+            span: Span::unknown(),
+            item: format!("(1..{}) | par-each -t 2 {{|_| sleep 50ns }} | ignore", n).to_string(),
```

Comment:
_shrugs at the limited number of core's on github workers_

Reply:
Do you think this should be removed or changed?, even if cores are low just allowing multiple commands to be run parallel allow some level of parallelism even if its run on 1 thread machine. I initially added this test as I saw 1thread `par-each` outperform the standard `each` for low sample counts.